scipy/linalg/tests/test_matfuncs.py

```diff
@@ -834,3 +834,61 @@ def test_expm_cond_fuzz(self):
             # eps times the condition number kappa.
             # In the limit as eps approaches zero it should never be greater.
             assert_array_less(p_best_relerr, (1 + 2*eps) * eps * kappa)
+
+
+class TestIsMatrix(object):
```

Comment:
These test are too simple and all integer based. There needs to be larger with varying entries (n=[30-100]), empty, zero and identity matrices to catch the edge cases.

Reply:
I have to be the bad news bringer but still the tests are too simple to be taken substantial and even they don't pass.

Reply:
Yes, indeed, I had some difficulties at first but now the tests and checks have all passed.

Reply:
I did the tests as you recommended.